compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/PropertyReferenceLowering.kt

```diff
@@ -0,0 +1,329 @@
+                        }
+                    }
+                    body = context.createIrBuilder(symbol).irBlockBody(startOffset, endOffset) {
+                        val superArgs = if (parameter == null) 0 else 1
+                        +irDelegatingConstructorCall(superClass.constructors.single { it.valueParameters.size == superArgs }).apply {
+                            parameter?.let { putValueArgument(0, irGet(parameter)) }
+                        }
+                        +IrInstanceInitializerCallImpl(startOffset, endOffset, referenceClass.symbol, context.irBuiltIns.unitType)
+                    }
+                }
+
+                fun buildOverride(method: IrSimpleFunction, build: IrBlockBodyBuilder.(List<IrValueParameter>) -> IrExpression) =
+                    buildFun {
+                        setSourceRange(expression)
+                        name = if (method.name == Name.special("<get-name>")) Name.identifier("getName") else method.name
+                        returnType = method.returnType
+                        visibility = method.visibility
+                        origin = referenceClass.origin
+                    }.apply {
+                        parent = referenceClass
+                        referenceClass.declarations.add(this)
+
+                        overriddenSymbols.add(method.symbol)
+                        dispatchReceiverParameter = referenceThis.copyTo(this)
+                        for (parameter in method.valueParameters)
+                            valueParameters.add(parameter.copyTo(this))
+                        body = context.createIrBuilder(symbol).irBlockBody(startOffset, endOffset) {
+                            +irReturn(build(valueParameters))
+                        }
+                    }
+
+                val receiverField = superClass.properties.single { it.name == Name.identifier("receiver") }.backingField!!
+                fun IrBuilderWithScope.setReceiversOn(call: IrCall, valueParameters: List<IrValueParameter>) {
+                    var index = 0
+                    call.dispatchReceiver = call.symbol.owner.dispatchReceiverParameter?.let {
+                        if (expression.dispatchReceiver != null)
+                            irGetField(irGet(referenceThis), receiverField)
+                        else
+                            irImplicitCast(irGet(valueParameters[index++]), it.type)
+                    }
+                    call.extensionReceiver = call.symbol.owner.extensionReceiverParameter?.let {
+                        if (expression.extensionReceiver != null)
+                            irGetField(irGet(referenceThis), receiverField)
+                        else
+                            irImplicitCast(irGet(valueParameters[index++]), it.type)
+                    }
+                }
+
+                buildOverride(superClass.properties.single { it.name == Name.identifier("name") }.getter!!) {
```

Comment:
Please replace `it.name == Name.identifier("xxx")` with `it.name.asString() == "xxx"`